### PR TITLE
'pack' freed  before  'yield' return  during channel message dispatch

### PR DESCRIPTION
**Pull request by niuys**
When there is a 'yield'  in a channel message dispatch function,  the 'pack'  is freed after the message dispatch,  but the invalid address is left for  'mc.close'  call in lua. Maybe there is better change method instead of adding new function.

### CODE REVIEW

File: lualib-src/lua-multicast.c [hunk 1]
```diff
@@ -83,7 +83,7 @@ mc_unpacklocal(lua_State *L) {
 	if (sz != sizeof(*pack)) {
 		return luaL_error(L, "Invalid multicast package size %d", sz);
 	}
-	lua_settop(L, 1);
+	lua_pushlightuserdata(L, (*pack));
```
Comment (cloudwu) [inline lualib-src/lua-multicast.c]: pack never free , memory leak here.

**Comment by cloudwu** [comment]
I see, I will find a better way to solve it.

**Comment by cloudwu** [comment]
I change the mc.close , please review the new patch.
thanks.

**Comment by niuys** [comment]
OK,graceful changes

**Pull request closed without merging**